content/jss_cart_actions.md

```diff
@@ -0,0 +1,233 @@
+
+The code for the `CartProvider` looks very similar to the `AuthProvider`:
+```
+import { AuthContext } from './AuthProvider';
+
+export const CartContext = React.createContext();
+
+async function getCart(token) {
+    return commerceRequest.get('api/carts/me', token);
+}
+
+async function addCartLine(token, line) {
+    await commerceRequest.put('api/carts/me/addline', token, JSON.stringify(line));
+}
+
+async function refreshCart(token, onUpdateCart) {
+    const cart = await getCart(token);
+    onUpdateCart(cart);
+    return cart;
+}
+
+export default function CartProvider({children}) {
+    const token = useContext(AuthContext);
+    const [cart, setCart] = useState(null);
+
+    const actions = {
+        addCartLine: async (line) => {
+            await addCartLine(token, line);
+            refreshCart(token, setCart);
+        }
+        ...
+    };
+
+    useEffect(() => {
+        refreshCart(token, setCart);
+    }, [token]);
+
+    const cartContext = {
+        data: cart,
+        actions: actions
+    };
+
+    return (
+        <CartContext.Provider value={cartContext}>
+            {children}
+        </CartContext.Provider>
+    );
+}
+```
+The only notable changes are:
+- a map that exposes all Cart actions, and contains actions that e.g. refresh the cart after an update
+- a token condition supplied to the `useEffect` which triggers a cart update when the `token` is refreshed
+
+
+# Create an Add To Cart button
+We can now create an *add to cart* button React component that uses the Cart context.
+
+```
+import React, {useContext} from 'react';
+import { CartContext } from './CartProvider';
+import { productCatalog } from '../../temp/config';
+
+const AddToCartButton = ({productId, variantId, ...other}) => {
+    const cart = useContext(CartContext);
+
+    const sellableItem = {
+        itemId: `${productCatalog}|${productId}|${variantId}`,
+        quantity: 1,
+        ...other
+    };
+
+    return (
+        <button className='addtocart' onClick={() => cart.actions.addCartLine(sellableItem)}>
+            Add to cart
+        </button>
+    );
+}
+```
+
+# Use it
+Last step is to actually use the button, for example by placing it on the `ProductSummary` component which is used for displaying products on a product cluster.
+As all required information is readily available on the `ProductSummary` component, adding the button is easy:
+```
+...
+
+const ProductSummary = ({path, displayName, description, imageId, productId, variantId}) => {
+    return <article className='productsummary'>
+      <Title path={path} displayName={displayName} />
+      <Description description={description} />
+      <ProductImage imageId={imageId}/>
+      <ProductPrice productId={productId} /> 
+      <AddToCartButton productId={productId} variantId={variantId} displayName={displayName} /> 
+    </article>
+}
+```
+
+Finally this results in having a button on the page, for example:
```

Comment:
Moeten we nog iets vertellen over JSS? Hoe de add-to-cart daar te gebruiken? En wat voor gevolgen het heeft voor disconnected mode bijv.